src/ducks/loan/LoanDetailsPage.spec.jsx

```diff
@@ -0,0 +1,212 @@
+        },
+        childContextTypes: {
+          t: PropTypes.func
+        }
+      })
+
+      expect(wrapper.html()).toBe('')
+    })
+  })
+})
+
+describe('characteristics section', () => {
+  describe('when at least one relevant data is defined', () => {
+    it('should render defined data', () => {
+      const wrapper = mount(
+        <CharacteristicsSection
+          account={{
+            loan: {
+              subscriptionDate: '2019-11-01',
+              maturityDate: '2020-11-01',
+              nbPaymentsLeft: 12,
+              nbPaymentsDone: 0
+            }
+          }}
+        />,
+        {
+          context: {
+            t: key => key,
+            f: date => date
+          },
+          childContextTypes: {
+            t: PropTypes.func,
+            f: PropTypes.func
+          }
+        }
+      )
+
+      expect(wrapper.find(Section)).toHaveLength(1)
+      expect(wrapper.find(Row)).toHaveLength(4)
+    })
+  })
+
+  describe('when no relevant data is defined', () => {
+    it('should not render', () => {
+      const wrapper = mount(<CharacteristicsSection account={{}} />, {
+        context: {
+          t: key => key
+        },
+        childContextTypes: {
+          t: PropTypes.func
+        }
+      })
+
+      expect(wrapper.html()).toBe('')
+    })
+  })
+})
+
+describe('credit reserve section', () => {
+  describe('when the account type is revolving credit', () => {
+    describe('when at least one relevant data is defined', () => {
+      it('should render defined data', () => {
+        const wrapper = mount(
+          <CreditReserveSection
+            account={{
+              loan: {
+                totalAmount: 10000,
+                availableAmount: 5000
+              },
+              type: 'RevolvingCredit'
+            }}
+          />,
+          {
+            context: {
+              t: key => key
+            },
+            childContextTypes: {
+              t: PropTypes.func
+            }
+          }
+        )
+
+        expect(wrapper.find(Section)).toHaveLength(1)
+        expect(wrapper.find(Row)).toHaveLength(2)
+      })
+    })
+
+    describe('when no relevant data is defined', () => {
+      it('should not render', () => {
+        const wrapper = mount(
+          <CreditReserveSection account={{ type: 'RevolvingCredit' }} />,
+          {
+            context: {
+              t: key => key
+            },
+            childContextTypes: {
+              t: PropTypes.func
+            }
+          }
+        )
+
+        expect(wrapper.html()).toBe('')
+      })
+    })
+  })
+
+  describe('when the account is not of type revolving credit', () => {
+    it('should not render', () => {
+      const wrapper = mount(
+        <CreditReserveSection account={{ type: 'RevolvingCredit' }} />,
+        {
+          context: {
+            t: key => key
+          },
+          childContextTypes: {
+            t: PropTypes.func
+          }
+        }
+      )
```

Comment:
Between each example the majority of lines are duplicated, the setup method pattern could be useful